### DISCUSSION TITLE
I'd like to actually extend models in the extensions, not override them

### DISCUSSION THREAD

**Discussion by laurenskling** [strapi/strapi#7887] [Legacy Topics]
I love the fact i can use the extensions folder. I'd like to extend a (plugin) model. Since models are JSON, and extensions are complete overrides, I need to completely copy the model into the extension, which makes it prone to upgrade errors. 

If I were to be able to write the model in JS, I could import the original and Object.assign my specific overrides. Or, if it would merge in my extension, as config files do. This would also work.

Having a way to actually extend the plugin models via extensions would be extremely powerful. Creating base models and specific extensions for specific use cases.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.
